### ISSUE TITLE
Some unicode chars not working properly when using 'utf8mb4'

### ISSUE THREAD

**Issue by jwshields** [aio-libs/aiomysql#237]
Thank you and my apologies in advance, I'm unsure what other way to word the title.

**Environment information:**
Python: 3.6.2
aiomysql: 0.0.11
MySQL: 5.7.20
Platform: Windows 10 (1709 16299.125)

**Problem**:
Some unicode characters are not encoding properly upon insertion to the table.

**My code**:

Connection:
```
self.mysqlcon = await aiomysql.create_pool(host=mysqlserver, port=mysqlport, user=mysqluser, minsize=1, maxsize=100, use_unicode=True, password=mysqlpw, db=mysqldb, autocommit=True, charset='utf8mb4')
```

Schema:
Note: when creating the schema, using MySQL WorkBench, I used the create schema wizard, and selected `utf8mb4 - default collation`
```
CREATE SCHEMA `db` DEFAULT CHARACTER SET utf8mb4 ;
```

Table Create: 
Note: tablename in the first line gets replaced by my python script
```
CREATE TABLE `db`.`tablename` (
  `id_messages` BIGINT NOT NULL AUTO_INCREMENT,
  `time` VARCHAR(45) NOT NULL,
  `user-name` LONGTEXT NOT NULL,
  `content` LONGTEXT NULL,
  PRIMARY KEY (`id_messages`));
```

Insert statement:
```
SQLQeury = """
        INSERT INTO `db`.`{0}_{1}_messages` (`time`, `user-name`, `content`)
        VALUES (%s, %s, %s);
        """
        NewQuery = SQLQeury.format(guildid, channelid)
        QueryData = (msgtime, username, content)
        tablename = str(guildid + "_" + channelid + "_messages")
        return NewQuery, tablename, QueryData
```

Insertion:
```
channelsqlcmd, tablename, querydata = self.bot.sql.statement_insert_channel_new(message)
await self.check_table("channel-new", tablename)
async with self.bot.mysqlcon.acquire() as conn:
    async with conn.cursor() as cursor:
        await cursor.execute(query=channelsqlcmd, args=querydata)
```

Results:
`SELECT * FROM db._messages;`
```
3, 2017-12-27 06:09:47.465798, boat, ????
4, 2017-12-27 06:09:53.782359, noodle, ?
5, 2017-12-27 06:09:56.413201, noodle, ❤
8, 2017-12-27 06:24:55.775261, noodle, ?
11, 2017-12-27 06:25:13.894280, noodle, ?
```

id 3, the last field should be (count 4) of 'FIRE' (U+1F525)
id 4, the last field should be (count 1) of 'Smiling Face with Open Mouth and Smiling Eyes' (U+1F604)
id 5, the last field is (count 1) of 'Red Heart' (U+2764)
id 8, the last field should be (count 1) of 'Sweat Droplets' (U+1F4A6)
id 11, the last field should be (count 1) of 'OK Hand' (U+1F44C)

Upon insertion, my Python console doesn't report errors.
Tested with two clients for getting results (MySQL Workbench, RazorSQL)

I've truncated some information for readability. (irrelevant Python lines, some of the columns returned from my queries)
I hope that I've provided some useful information;
I've tried to include as much as I can, please let me know if you need anything else.

**Comment by jwshields** [comment]
I've been thinking more on this. I can't tell if this is a problem with aiomysql, pymysql, or mysql itself.
I have a question though.
1. When creating the schema/table/column, what should the charset and collation be? What should I be setting in aiomysql on the connect params?
I've got my current config for aiomysql set with the charset of utf8mb8 because selecting utf16 or utf32 produces an error and will not connect to the DB. (`aiomysql.create_pool([other params], charset='utf8mb4')`)
If I set the aiomysql charset param to utf8 - it produces an error on insert along the lines of this:
```
`pymysql.err.InternalError: (1366, "Incorrect string value: '\\xF0\\x9F\\x94\\xA5\\xF0\\x9F...' for column 'content' at row 1")`
```

I've tested with aiomysql on `utf8mb4` and creating the schema/table/column with `utf8mb4 - utf8mb4_bin`, `utf8mb4 - utf8mb4_general_ci`, `utf8mb4 - default collation`, `utf8mb4 - utf8mb4_unicode_ci`, `utf32 - utf32_bin`, `utf32 - utf32_unicode_ci`

Each of these charsets that I listed, do not produce any errors in my Python console when inserting into the DB, but when using MySQL Workbench, the "emoji" unicode characters (fire, ok hand, etc) - show up as question marks, as detailed in my initial post.

**Comment by jwshields** [comment]
My apologies for all the spam on this issue.
It appears the problem was with MySQL Workbench (and the other MySQL DB Browser as well) - where MySQL WB was set to using `utf8` - and the server was sending back `utf8mb4`

I ran this query from my client:
```
SET NAMES 'utf8mb4' COLLATE 'utf8mb4_unicode_ci'
```
And now my queries are showing the proper text.
Closing ticket.
There was never a problem with aiomysql, mysql, nor pymysql storing the wrong characters.
